### ISSUE TITLE
Support multi-line labels

### ISSUE THREAD

**Issue by mramato** [AnalyticalGraphicsInc/cesium#2402]
This came up [on the forum](https://groups.google.com/d/msg/cesium-dev/HMl8gbzGMlw/o9Lx6bZlOPoJ) a while ago and the thread was recently revived to see if we had support yet.

I think it should be possible to do by detecting new line breaks and calculating a y offset in addition to the x offset we already calculate.  @shunter is this as easy as it sounds or am I missing something?

**Comment by andre-nunes** [comment]
Hey, I'd like to implement this and would like more details.

After a quick look I think I'd need to change ``LabelCollection.js``, but what exactly? Do I add the y-offset calculation to the ``repositionAllGlyphs`` function? What else would need to be changed? 

Let me know, thanks.

**Comment by mramato** [comment]
I think that's exactly what you need to do.  There might be more to it than that, but we probably won't know until we try.  It might turn out to be pretty simple.

**Comment by andre-nunes** [comment]
It seems that @abwood had a working initial implementation. I hope there's no problem in me using this commit https://github.com/abwood/cesium/commit/689a166f6d0fb265a1fcddbeac098a9343734a64 as a starting point, right now it looks like this:

![multiline](https://cloud.githubusercontent.com/assets/3066457/6496233/611fe914-c2c9-11e4-902e-6528202a2938.png)

Should the label be completely above the billboard? Looks like the computed y offset is not enough.

**Comment by mramato** [comment]
Awesome, yes, starting with @abwood's code is fine in this case.

Where the text ends up is dependant on the vertical origin setting. Look at the documentation for vertical origin here: http://cesiumjs.org/Cesium/Build/Documentation/Label.html#verticalOrigin there's also a useful picture.  


The offset by distance stuff is probably getting in the way here, so I wouldn't use it for your testing. Here's some example code you can use instead.  If the verticalOrigin is "bottom" text should grow "up" and if the origin is top, text should grow "down".  And when origin is center, text should be at the center of position.

```javascript
var viewer = new Cesium.Viewer('cesiumContainer');

viewer.entities.add({
    position : Cesium.Cartesian3.fromDegrees(0,0,0),
    point : {
        pixelSize : 5
    },
    label : {
        text : 'Center'
    }
});

viewer.entities.add({
    position : Cesium.Cartesian3.fromDegrees(0,0,0),
    point : {
        pixelSize : 5
    },
    label : {
        text : 'Top',
        verticalOrigin : Cesium.VerticalOrigin.TOP
    }
});

viewer.entities.add({
    position : Cesium.Cartesian3.fromDegrees(0,0,0),
    point : {
        pixelSize : 5
    },
    label : {
        text : 'BOTTOM',
        verticalOrigin : Cesium.VerticalOrigin.BOTTOM
    }
});

viewer.zoomTo(viewer.entities);
```

**Comment by andre-nunes** [comment]
It seems to be working pretty well:
![working](https://cloud.githubusercontent.com/assets/3066457/6586866/464cfbb2-c775-11e4-9475-11168bf69df2.png)

Should I write some unit tests? Can you give me an example on how to test this one?

Thanks

**Comment by mramato** [comment]
Awesome.  You can probably reach into the `LabelCollection` and verify that it generated the billboards with the expected pixelOffsets.  Feel free to open the PR for review whenever you like and mention that you still need to add tests.

**Comment by pjcozzi** [comment]
Fixed in #4306.
